src/aggregation/mod.rs

```diff
@@ -271,6 +271,22 @@ pub enum Key {
     /// `f64` key
     F64(f64),
 }
+impl Eq for Key {}
+impl std::hash::Hash for Key {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        core::mem::discriminant(self).hash(state);
```

Comment:
This won't work well if someone indexes different values of `f64::NAN` and then does an aggregation on it. 
Do we care about this use case?

Reply:
Let's forbid NaN entirely.

Reply:
TIL about Discriminant